Remove macos-12 buildbots

bnoordhuis opened the pull request:
GitHub is deprecating and removing them. We also test macos-14 so I opted to remove the buildbots instead of upgrading them to a newer macOS version.

saghul commented:
I thint the macOS-14 image it arm64 only. If we want to test x64 too we should add the -large variant IIRC. Other than that, LGTM!

bnoordhuis commented:
I think the gcc48 buildbot is nearing its end of life, too...
```
The repository will be downloaded using the GitHub REST API
To create a local Git repository instead, add Git 2.18 or higher to the PATH
Error: Input 'submodules' not supported when falling back to download using the GitHub REST API. To create a local Git repository instead, add Git 2.18 or higher to the PATH.
```

bnoordhuis commented:
> If we want to test x64 too we should add the -large variant

I don't know what that means :grin:

saghul commented:
> > If we want to test x64 too we should add the -large variant
> 
> 
> 
> I don't know what that means :grin:

See: https://github.com/actions/runner-images

macos-14 is arm64 and macos-14-large is x64. macos-12 was x64 (there was no arm64 option for macos 12 IIRC).

saghul commented:
> I think the gcc48 buildbot is nearing its end of life, too...
> 
> ```
> 
> The repository will be downloaded using the GitHub REST API
> 
> To create a local Git repository instead, add Git 2.18 or higher to the PATH
> 
> Error: Input 'submodules' not supported when falling back to download using the GitHub REST API. To create a local Git repository instead, add Git 2.18 or higher to the PATH.
> 
> ```

We could always clone in a new runner and run the tests in a container. But if / when it becomes too much hassle: 🪓

bnoordhuis commented:
macos-14-large: "This job failed", no further explanation

Are large instances a paying customer thing?

saghul commented:
> macos-14-large: "This job failed", no further explanation
> 
> 
> 
> Are large instances a paying customer thing?

Oh could be indeed! We are good then!

bnoordhuis committed:
Remove macos-12 buildbots

GitHub is deprecating and removing them. We also test macos-14 so I
opted to remove the buildbots instead of upgrading them to a newer
macOS version.
